src/GitRepository.cc

```diff
@@ -0,0 +1,416 @@
+/**
+ * \file   GitRepository.cc
+ * \author Sven-Jannik Wöhnert
+ * \date   Created on March 20, 2023
+ * \brief  Wrapper for C-Package libgit2
+ *
+ * \copyright Copyright 2023 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+
+
+#include <git2.h>
+#include "taskolib/GitRepository.h"
+#include "taskolib/exceptions.h"
+#include "gul14/cat.h"
+#include <iostream>
+#include <vector>
+
+namespace task {
+
+
+GitRepository::GitRepository(std::filesystem::path file_path)
+{
+    //init libgit library
+    git_libgit2_init();
+ 
+    //init member variables
+    repo_path_ = file_path;
+
+    // if repository does not exist
+    repo_ = repository_open(repo_path_);
+    if (repo_.get() == nullptr) 
+    {
+        // initialize everything
+        init(file_path);
+    }
+    else
+    {
+        // intialize the signature
+        my_signature_ = signature_default(repo_.get());
+        if (my_signature_.get() == nullptr) my_signature_ = signature_new("Taskomat", "taskomat@desy.de", 123456789, 0);
+    }
+}
+
+
+GitRepository::~GitRepository()
+{
+    repo_.reset();
+    my_signature_.reset();
+    git_libgit2_shutdown();
+}
+
+std::string GitRepository::get_last_commit_message()
+{
+    const git_commit *commit = get_commit();
```

Comment:
I guess this is a memory leak? `get_commit()` returns a raw pointer, but no one ever calls free() on it?